staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/streaming.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package json
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"maps"
+	"slices"
+	"sort"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/conversion"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func streamingEncode(obj runtime.Object, w io.Writer) (bool, error) {
+	typeMeta, listMeta, items, err := getListMeta(obj)
+	if err == nil {
+		return true, streamingEncodeList(w, typeMeta, listMeta, items)
+	}
```

Comment:
Does anything prevent an aggregated apiserver from serving a list type with the expected fields _and_ a MarshalJSON implementation? I wasn't able to find an example of this in the wild, but it may be worth ensuring that the MarshalJSON implementation is not bypassed in that case.

Reply:
I think @benluddy is asking about a case where the *list* type implements MarshalJSON

Reply:
> the _list_ type implements MarshalJSON

Yes, I'd expect such a type not to be streamed.

Reply:
Ok, sorry for missing it. Yes,  you are right I didn't consider aggregated apiserver case with it's own types that we don't control. I as K8s builtin types don't implement MarshalJson we just check if provided type does and skip streaming. We can add a test for that.

Reply:
Added testing of MarshallJSON to KEP test plan.